### ISSUE TITLE
vistit: Retry on ENOTFOUND errors

### ISSUE THREAD

**Issue by testerez** [cypress-io/cypress#4424]
### Current behavior:

When using `cy.vistit`, if the DNS fails to resolve, we get: `Error: getaddrinfo ENOTFOUND`

Example:
![Screen Shot 2019-06-10 at 10 37 09 AM](https://user-images.githubusercontent.com/815236/59203009-b7a02c80-8b6b-11e9-9931-e1c56af328e7.png)

### Desired behavior:

In our CI, the DNS is sometimes flaky. Unfortunately, we can't really fix that yet. So it would be nice if cypress offered a way to retry the `visit` command when we get this kind of errors.

### Versions

cypress@3.3.1

**Comment by jennifer-shehane** [comment]
We recently implemented automatic retrying of `cy.visit()` in these situations in [Cypress 3.3.0](https://on.cypress.io/changelog#3-3-0), so it has already retried and failed at this point. 

>Requests that go through `cy.visit()` or `cy.request()` are now automatically retried under the hood when the request fails. This more closely reflects Chrome’s default behavior on requests. Addresses #4013, #898 and #1013.

@flotwig We mention that we retried the `cy.visit()` in the CLI log when it fails on `baseUrl`, shouldn't we also mention we retried the `cy.visit()` in the Command Log here?

You can implement rerunning the commands/tests when they fail using the `cypress-plugin-retires` mentioned here. https://github.com/cypress-io/cypress/issues/1313#issuecomment-460938886 

I'm afraid this is all we can recommend in this situation.

**Comment by testerez** [comment]
From a quick test, it does not look like cypress retries in this specific situation. You can easily reproduce with:
```
it('will fail instantly with ENOTFOUND', () => {
  cy.visit('http://foo.localhost');
});
```
When I try this, Cypress fails instantly (0.21s):

<img width="596" alt="Screen Shot 2019-06-11 at 6 49 26 AM" src="https://user-images.githubusercontent.com/815236/59266244-300ef800-8c15-11e9-9920-aebef508ee24.png">

**Comment by flotwig** [comment]
We initially left out retrying on ENOTFOUND because we thought this scenario didn't exist in the wild (flaky DNS???), but it can definitely be added.

**Comment by testerez** [comment]
Cool thank you @flotwig! To be honest I'm not entirely sure why we have this issue. But it happens when our CI deploys an ephemeral to test against. We do wait until the deployment is ready before starting the Cypress tests but sometimes we still get this intermittent ENOTFOUND error. Maybe the record is not yet propagated everywhere. Anyway, if Cypress could just retry when we get this error it would be a quick fix for us.

**Comment by flotwig** [comment]
Actually found another instance of this DNS flakiness, so I guess it's not as rare as we thought: #1755

**Comment by pardeeprj90** [comment]
Error: getaddrinfo ENOTFOUND vfolpuat.ctdi.com
![image](https://user-images.githubusercontent.com/37590015/100450233-4c3d7800-30db-11eb-83d2-f2213507525e.png)

still, issue exists, the same code was working previously fine.

**Comment by jennifer-shehane** [comment]
This issue will be closed to further comment as the exact issue here was resolved and tested. 

@pardeeprj90 If you're experiencing a bug similar to this in Cypress, please [open a new issue](https://github.com/cypress-io/cypress/issues/new/choose) with a fully reproducible example that we can run. There may be a specific edge case with the issue that we need more detail to fix.
